### ISSUE TITLE
Algorithm R yielding improper results

### ISSUE THREAD

**Issue by kartva** [DesmondWillowbrook/rs-reservoir-sampling#1]
Rerunning `cargo run --example u32` 6 times yields the following results:
```
With Algorithm L: [63, 1, 2, 3, 96, 5, 34, 7, 8, 9]
With Algorithm R: [83, 96, 80, 88, 95, 56, 89, 97, 93, 99]

With Algorithm L: [0, 63, 2, 3, 79, 96, 34, 7, 8, 9]
With Algorithm R: [95, 97, 96, 99, 71, 98, 94, 91, 81, 88]

With Algorithm L: [51, 67, 2, 84, 4, 5, 6, 7, 8, 36]
With Algorithm R: [96, 97, 99, 86, 95, 94, 69, 91, 98, 93]

With Algorithm L: [0, 50, 2, 68, 4, 87, 35, 7, 8, 9]
With Algorithm R: [90, 98, 97, 96, 78, 85, 66, 87, 92, 99]

With Algorithm L: [0, 1, 2, 3, 4, 66, 6, 21, 83, 9]
With Algorithm R: [96, 87, 95, 98, 47, 97, 82, 99, 90, 92]

With Algorithm L: [48, 1, 96, 3, 4, 5, 79, 7, 8, 63]
With Algorithm R: [89, 79, 88, 96, 90, 98, 75, 95, 99, 93]
```

Clearly Algorithm R is biased towards the values at the end.

**Comment by kartva** [comment]
Algorithm R is temporarily being removed from the API until it is fixed.

**Comment by kartva** [comment]
Has been fixed as of `v0.3.0`!
